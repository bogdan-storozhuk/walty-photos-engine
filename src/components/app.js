import React from 'react';
import { Route, Switch } from 'react-router-dom';
import 'bootstrap/dist/css/bootstrap.min.css';

import { MainPage, PhotoPageContainer } from '.';

const App = () => (
  <Switch>
    <Route path="/" component={MainPage} exact />
    <Route
      path="/photo-page/:id"
      render={({ match }) => {
        const { id } = match.params;
        return <PhotoPageContainer photoId={id} />;
      }}
      exact
    />
    <Route
      path="/:tag1?/:tag2?/:tag3?/:tag4?/:tag5?/:tag6?/:tag7?/:tag8?/:tag9?/:tag10?"
      render={({ match }) => <MainPage tags={match.params} />}
    />
  </Switch>
);

App.propTypes = {};

export default App;
