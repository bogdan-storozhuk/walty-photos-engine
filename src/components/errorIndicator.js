import React from 'react';

const ErrorIndicator = () => <div>Error!</div>;

ErrorIndicator.propTypes = {};

export default ErrorIndicator;
