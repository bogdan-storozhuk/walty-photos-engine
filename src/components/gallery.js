import React from 'react';
import { useSelector } from 'react-redux';
import PropTypes from 'prop-types';

import { GalleryItem } from './';
import { Spinner } from './';
import { ErrorIndicator } from './';

const Gallery = ({ photos }) => (
  <div className="row mt-3">
    {photos.map((photo) => {
      return <GalleryItem key={photo.id} photo={photo} />;
    })}
  </div>
);

const GalleryContainer = () => {
  const photos = useSelector((state) => state.photoReducer.photos),
    loading = useSelector((state) => state.photoReducer.loading),
    error = useSelector((state) => state.photoReducer.error);
  if (loading) {
    return <Spinner />;
  }

  if (error) {
    return <ErrorIndicator />;
  }

  return <Gallery photos={photos} />;
};

Gallery.propTypes = {
  photos: PropTypes.arrayOf(
    PropTypes.shape({
      largeImageURL: PropTypes.string,
      id: PropTypes.number,
    })
  ),
};

export default GalleryContainer;
