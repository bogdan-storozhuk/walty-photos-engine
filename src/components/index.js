import App from './app';
import ErrorBoundry from './errorBoundry';
import ErrorIndicator from './errorIndicator';
import GalleryContainer, { Gallery } from './gallery';
import GalleryItem from './galleryItem';
import LastSearchesPanel from './lastSearchesPanel';
import MainPage from './mainPage';
import SearchPanel from './searchPanel';
import Spinner from './spinner';
import Tag from './tag';
import PhotoPageContainer, { PhotoPage } from './photoPage';

export {
  App,
  MainPage,
  SearchPanel,
  LastSearchesPanel,
  GalleryContainer,
  Gallery,
  GalleryItem,
  Tag,
  ErrorBoundry,
  ErrorIndicator,
  Spinner,
  PhotoPageContainer,
  PhotoPage,
};
