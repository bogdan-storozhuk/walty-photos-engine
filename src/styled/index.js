import GalleryItemContainer from './galleryItemContainer';
import GalleryItemImage from './galleryItemImage';
import TagSpan from './tagSpan';
import StyledLink from './styledLink';

export { GalleryItemContainer, GalleryItemImage, TagSpan, StyledLink };
