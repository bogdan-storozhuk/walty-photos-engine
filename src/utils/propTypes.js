import PropTypes from 'prop-types';

export const galleryPropTypes = {
  photos: PropTypes.arrayOf(
    PropTypes.shape({
      largeImageURL: PropTypes.string,
      id: PropTypes.number,
    })
  ),
};

export const galleryItemPropTypes = {
  photo: PropTypes.shape({
    largeImageURL: PropTypes.string,
    id: PropTypes.number,
  }),
};

export const tagPropTypes = {
  name: PropTypes.string,
};

export const photoPagePropTypes = {
  singlePhoto: PropTypes.shape({
    id: PropTypes.number,
    downloads: PropTypes.number,
    favorites: PropTypes.number,
    likes: PropTypes.number,
    views: PropTypes.number,
    tags: PropTypes.string,
    largeImageURL: PropTypes.string,
  }),
};

export const photoPageContainerPropTypes = {
  photoId: PropTypes.string,
};

export const mainPagePropTypes = {
  tags: PropTypes.shape({
    tag1: PropTypes.string,
    tag2: PropTypes.string,
    tag3: PropTypes.string,
    tag4: PropTypes.string,
    tag5: PropTypes.string,
    tag6: PropTypes.string,
    tag7: PropTypes.string,
    tag8: PropTypes.string,
    tag9: PropTypes.string,
    tag10: PropTypes.string,
  }),
};
