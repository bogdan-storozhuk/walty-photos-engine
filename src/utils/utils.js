import checkPropTypes from 'check-prop-types';

const deleteUndefinedValuesFromObject = (obj) => {
  for (var propName in obj) {
    if (obj[propName] === null || obj[propName] === undefined) {
      delete obj[propName];
    }
  }
};

const transformObjectIntoArray = (obj) => {
  return Object.values(obj).map((item) => ({
    name: item,
    id: Date.now() + Math.floor(Math.random() * 1000),
  }));
};

const generateRandomId = () => Date.now() + Math.floor(Math.random() * 1000);

const checkProps = (component, expectedProps, propTypes) => {
  const propsErr = checkPropTypes(
    propTypes,
    expectedProps,
    'props',
    component.name
  );
  return propsErr;
};

export {
  deleteUndefinedValuesFromObject,
  transformObjectIntoArray,
  generateRandomId,
  checkProps,
};
