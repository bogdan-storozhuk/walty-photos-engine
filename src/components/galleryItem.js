import React from 'react';
import PropTypes from 'prop-types';

import { GalleryItemContainer, GalleryItemImage } from '../styled/';

const GalleryItem = ({ photo }) => (
  <GalleryItemContainer className="col-lg-3 col-md-4 col-6 mb-4">
    <GalleryItemImage
      src={photo.largeImageURL}
      alt={photo.id}
    ></GalleryItemImage>
  </GalleryItemContainer>
);

GalleryItem.propTypes = {
  photo: PropTypes.shape({
    largeImageURL: PropTypes.string,
    id: PropTypes.number,
  }),
};

export default GalleryItem;
