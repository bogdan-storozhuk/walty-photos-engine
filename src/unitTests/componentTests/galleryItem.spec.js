import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { GalleryItem } from '../../components/';
import { galleryItemPropTypes } from '../../utils/propTypes';
import { checkProps } from '../../utils/utils';
import { GalleryItemContainer, GalleryItemImage } from '../../styled/';

configure({ adapter: new Adapter() });

describe('GalleryItem Component', () => {
  describe('Checking PropTypes', () => {
    it('Should not throw a warning', () => {
      const expectedProps = {
        largeImageURL: 'some url',
        id: 1,
      };
      const propsError = checkProps(
        GalleryItem,
        expectedProps,
        galleryItemPropTypes
      );
      expect(propsError).toBeUndefined();
    });
  });

  describe('GalleryItem Component initial', () => {
    const props = {
      photo: {
        largeImageURL: 'some url',
        id: 1,
      },
    };
    const galleryItem = shallow(<GalleryItem {...props} />);

    it('Should render without errors', () => {
      expect(galleryItem.find(GalleryItemContainer)).toHaveLength(1);
    });

    it('Should render an imag', () => {
      expect(galleryItem.find(GalleryItemImage)).toHaveLength(1);
    });

    it('Should render properly', () => {
      expect(galleryItem).toMatchSnapshot();
    });
  });
});
