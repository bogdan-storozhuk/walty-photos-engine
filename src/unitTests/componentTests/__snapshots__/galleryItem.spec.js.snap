// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryItem Component GalleryItem Component initial Should render properly 1`] = `ShallowWrapper {}`;
